.travis.yml

```diff
@@ -1,64 +1,169 @@
+#    variables COVERITY_SCAN_TOKEN and COVERITY_SCAN_NOTIFICATION_EMAIL.  
+# 6. Enable pull request builds in your boostorg/<library> account.
+#
+# That's it - the script will do everything else for you.
 
 sudo: false
+dist: trusty
+language: cpp
 
-python: "2.7"
+env:
+  global:
+  # see: http://www.boost.org/build/doc/html/bbv2/overview/invocation.html#bbv2.overview.invocation.properties
+  # to use the default for a given environment, comment it out; recommend you build debug and release however..
+  # - B2_ADDRESS_MODEL=address-model=64,32
+  # - B2_LINK=link=shared,static
+  # - B2_THREADING=threading=multi,single
+    - B2_VARIANT=variant=release,debug
 
+install:
+  - export SELF=`basename $TRAVIS_BUILD_DIR`
+  - cd ..
+  - git clone -b $TRAVIS_BRANCH --depth 1 https://github.com/boostorg/boost.git boost-root
+  - cd boost-root
+  - git submodule update -q --init tools/boostdep
+  - git submodule update -q --init tools/build
+  - git submodule update -q --init tools/inspect
+  - cp -r $TRAVIS_BUILD_DIR/* libs/$SELF
+  - export BOOST_ROOT="`pwd`"
+  - export PATH="`pwd`":$PATH 
+  - python tools/boostdep/depinst/depinst.py $SELF --include example
+  - ./bootstrap.sh
+  - ./b2 headers
+
+addons:
+  apt:
+    packages:
+      - binutils-gold
+      - gdb
+      - libc6-dbg
+        
 branches:
   only:
-    - master
     - develop
-    - /feature\/.*/
-
-env:
-  matrix:
-    - BOGUS_JOB=true
-
-matrix:
-
-  exclude:
-    - env: BOGUS_JOB=true
+    - master
+    
+script:
+  - cd libs/$SELF
+  - ci/build.sh
 
+jobs:
   include:
+    #################### Jobs to run on every pull request ####################
+    - os: linux
+      env: 
+        - COMMENT="C++03"
+        - TOOLSET=gcc,gcc-7
+      addons:
+        apt:
+          packages:
+            - g++-7
+          sources:
+            - ubuntu-toolchain-r-test
     - os: linux
-      compiler: g++
-      env: TOOLSET=gcc CXXSTD=03,11
+      env: 
+        - COMMENT="C++11"
+        - TOOLSET=clang
+        - CXXSTD=11
+      addons:
+        apt:
+          packages:
+            - g++-7
+          sources:
+            - ubuntu-toolchain-r-test
+    - os: linux
+      env: 
+        - COMMENT=valgrind
+        - TOOLSET=clang 
+        - B2_VARIANT=variant=debug
+        - TESTFLAGS=testing.launcher=valgrind
+      addons:
+        apt:
+          packages:
+            - clang-5.0
+            - libstdc++-7-dev
+            - valgrind
+          sources:
+            - llvm-toolchain-trusty-5.0
+            - ubuntu-toolchain-r-test
+
+    - os: linux
+      env:
+        - COMMENT=cppcheck
+      script:
+        - libs/$SELF/ci/cppcheck.sh
 
     - os: linux
-      compiler: clang++
-      env: TOOLSET=clang CXXSTD=03,11
+      env:
+        - COMMENT=UBSAN
+        - B2_VARIANT=variant=debug
+        - TOOLSET=gcc-7
+        - CXXFLAGS="cxxflags=-fno-omit-frame-pointer cxxflags=-fsanitize=undefined"
```

Comment:
You have to use `-fno-sanitize-recover=undefined` here; without it, the sanitizer prints messages to stderr, but the test passes and nobody's any the wiser. :-)

Reply:
Thanks!  I will change this too.

Reply:
@pdimov thanks for catching this; I applied it to the other repositories where I run UBSAN and it uncovered something in serialization, and as you can see in the failed build uncovered a few things in DateTime too.